python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -3069,6 +3069,53 @@ def foo(x: int):
     assertThat(yType.typeSource()).isSameAs(TypeSource.TYPE_HINT);
   }
 
+  public static Stream<Arguments> binaryExpressionsSource() {
+    return Stream.of(
+      Arguments.of("""
+        1 + 2
+        """, INT_TYPE),
+      Arguments.of("""
+        1 - 2
+        """, INT_TYPE),
+      Arguments.of("""
+        1 * 2
+        """, INT_TYPE),
+      Arguments.of("""
+        1 / 2
+        """, INT_TYPE),
+      Arguments.of("""
+        '1' + '2'
+        """, STR_TYPE),
+      Arguments.of("""
+        a = 1
+        b = 2
+        a + b
+        """, INT_TYPE),
+      Arguments.of("""
+        a = 1
+        b = 2
+        c = a - b
+        c
+        """, INT_TYPE),
+      Arguments.of("""
+        try:
+          ...
+        except:
+          ...
+        a = 1
+        b = 2
+        c = a - b
+        c
+        """, INT_TYPE)
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("binaryExpressionsSource")
+  void binaryExpressionTest(String code, PythonType expectedType) {
+    assertThat(lastExpression(code).typeV2()).isInstanceOf(ObjectType.class).extracting(PythonType::unwrappedType).isEqualTo(expectedType);
```

Comment:
It would be nice to have negative tests. For example to check if `1 + 2.0` is `UNKNOWN`.